github/client.go

```diff
@@ -33,6 +33,16 @@ func (c *Client) GetGithubRepoFromRawURL(rawURL string) (svcmodel.GithubRepo, er
 	return repo, nil
 }
 
+func (c *Client) GenerateGitTagURL(repo svcmodel.GithubRepo, tagName string) (url.URL, error) {
```

Comment:
Accepting svcmodel.GithubRepo (instead of url)

Reply:
Btw, here as well, as I once pointed out about having separate type for `userID` (= dumb wrapper around `uuid.UUID`), here a part of me also asks for having a type `githubURL`.
If time allows, would be almost a banger for the thesis / project.